keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1224 @@
+        be added later.
+
+      * `pod.spec.resources.requests[memory/cpu] < sum(pod.spec.containers[*].resources.requests[memory/cpu])` is invalid.
+
+          This must result in an error as it is semantically inconsistent. Pod-level requests are meant to represent the minimum amount of resources needed for 
+          all the containers within a pod to function correctly. If a pod’s request is lower than the sum of its container requests, it creates a logical 
+          contradiction: the pod as a whole requires less resources than the sum of its parts.
+
+          Besides, the scheduler will use pod requests to determine suitable nodes for placement. If a pod’s request underestimates its actual resource needs
+          (represented by the sum of container requests), it could be scheduled on a node that cannot provide enough resources, leading to failures and 
+          performance degradation.
+
+  2. limits
+      * `pod.spec.resources.limits[memory/cpu] >= sum(pod.spec.containers[*].resources.limits[memory/cpu])` is valid.
+
+          Pod-level resource limits are allowed to be greater than the sum of container level resource limits to account for ephemeral containers that could be added 
+          later. Setting a resource limit on the pod in its cgroup settings will not reserve any resource and hence not affect the other workloads running on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < sum(pod.spec.containers[*].resources.limits[memory/cpu])` is valid.
+        
+        For example: consider a pod with multiple containers:
+          1. Container 1: Moderate CPU and memory usage, but occasional spikes.
+          2. Container 2: Low and consistent resource usage.
+
+
+          ```yaml
+          apiVersion: v1
+          kind: Pod
+          ...
+          spec:
+            resources:
+              requests:
+                memory: 100Mi
+                cpu: 1
+              limits:
+                memory: 400Mi
+                cpu: 2.5
+            containers:
+            - name: container-1
+                resources:
+                  requests:
+                    memory: 60Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 300Mi
+                    cpu: 2  
+            - name: container-2
+                resources:
+                  requests:
+                    memory: 40Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 200Mi
+                    cpu: 1 
+          ```
+
+          The pod-level limit will serve as an aggregate ceiling. This will cover the scenario where the sum of peak usage for individual containers in a pod is high, but these
+          peaks don’t necessarily occur simultaneously.
+
+          If we simply set the pod-level limit as the sum of the individual container limits, we might overallocate resources. The pod might never actually use all those resources
```

Comment:
FWIW, I think this is the most interesting aspect of the whole KEP. 

If I make a pod which requests.memory = 8Gi and limits.memory = 8Gi, and I have 2 containers which each requests.memory=4Gi and limits.memory = 5Gi, I think think the meaning is clear and unambiguous.  I want guaranteed QOS.  I will not over-commit the machine.  I am asserting that my own containers might burst by 25%, but I think they will rarely burst at the same time.  And if they do, I should expect reclaim and OOM inside my pod.

Reply:
Another flavor of the same: https://github.com/kubernetes/enhancements/pull/4678/files#r1645182074